.travis.yml

```diff
@@ -14,22 +14,33 @@ matrix:
     - {python: 3.5, env: TOXENV=py35-poll}
     - {python: 3.5, env: TOXENV=py35-selects}
 
-    - {python: 3.6, env: TOXENV=py36-epolls}
-    - {python: 3.6, env: TOXENV=py36-poll}
-    - {python: 3.6, env: TOXENV=py36-selects}
+    - {python: 3.6, env: TOXENV=py36-epolls-dnspython1}
+    - {python: 3.6, env: TOXENV=py36-poll-dnspython1}
+    - {python: 3.6, env: TOXENV=py36-selects-dnspython1}
 
-    - {python: 3.7, sudo: required, dist: xenial, env: TOXENV=py37-epolls}
-    - {python: 3.7, sudo: required, dist: xenial, env: TOXENV=py37-poll}
-    - {python: 3.7, sudo: required, dist: xenial, env: TOXENV=py37-selects}
+    - {python: 3.6, env: TOXENV=py36-selects-dnspython2}
+
+    - {python: 3.7, sudo: required, env: TOXENV=py37-epolls-dnspython1}
+    - {python: 3.7, sudo: required, env: TOXENV=py37-poll-dnspython1}
+    - {python: 3.7, sudo: required, env: TOXENV=py37-selects-dnspython1}
+
+    - {python: 3.8, sudo: required, env: TOXENV=py38-epolls-dnspython1}
+    - {python: 3.8, sudo: required, env: TOXENV=py38-poll-dnspython1}
+    - {python: 3.8, sudo: required, env: TOXENV=py38-selects-dnspython1}
+
+    - {python: 3.8, env: TOXENV=py38-selects-dnspython2}
 
     - {python: pypy, env: TOXENV=pypy-epolls}
     - {python: pypy, env: TOXENV=pypy-poll}
     - {python: pypy, env: TOXENV=pypy-selects}
 
     - {python: 3.6, env: TOXENV=ipv6}
   allow_failures:
+    - python: 2.7
```

Comment:
@temoto could you please review https://github.com/eventlet/eventlet/pull/640  - this PR you are reviewing sits on top of that one so that the CI is sane, but that is where it is done.